pkg/services/rpcsrv/server_test.go

```diff
@@ -2275,96 +2275,223 @@ func TestSubmitOracle(t *testing.T) {
+			compareTx(t, tx, actual)
 		}
-		p := &payload.P2PNotaryRequest{
-			MainTransaction:     mainTx,
-			FallbackTransaction: fallbackTx,
-			Witness: transaction.Witness{
-				InvocationScript:   []byte{1, 2, 3},
-				VerificationScript: []byte{7, 8, 9},
-			},
+		t.Run("mainTx verbose", func(t *testing.T) {
+			checkGetTxVerbose(t, notaryRequest1.MainTransaction)
+		})
+		t.Run("fallbackTx verbose", func(t *testing.T) {
+			checkGetTxVerbose(t, notaryRequest1.FallbackTransaction)
+			checkGetTxVerbose(t, notaryRequest2.FallbackTransaction)
+		})
+
+		checkGetTxBytes := func(t *testing.T, tx *transaction.Transaction) {
+			req := fmt.Sprintf(rpcTx, tx.Hash().StringLE(), 0)
+			body := doRPCCallOverHTTP(req, httpSrv.URL, t)
+			res := checkErrGetResult(t, body, false, 0)
+
+			var s string
+			err := json.Unmarshal(res, &s)
+			require.NoErrorf(t, err, "could not parse response: %s", res)
+			txBin, err := testserdes.EncodeBinary(tx)
+			require.NoError(t, err)
+			expected := base64.StdEncoding.EncodeToString(txBin)
+			assert.Equal(t, expected, s)
 		}
-		bytes, err := p.Bytes()
-		require.NoError(t, err)
-		str := fmt.Sprintf(`"%s"`, base64.StdEncoding.EncodeToString(bytes))
-		runCase(t, true, neorpc.ErrVerificationFailedCode, str)(t)
-	})
-	t.Run("valid request", func(t *testing.T) {
-		sender := testchain.PrivateKeyByID(0) // owner of the deposit in testchain
-		p := createValidNotaryRequest(chain, sender, 1)
-		bytes, err := p.Bytes()
-		require.NoError(t, err)
-		str := fmt.Sprintf(`"%s"`, base64.StdEncoding.EncodeToString(bytes))
-		runCase(t, false, 0, str)(t)
+		t.Run("mainTx bytes", func(t *testing.T) {
+			checkGetTxBytes(t, notaryRequest1.MainTransaction)
+		})
+		t.Run("fallbackTx bytes", func(t *testing.T) {
+			checkGetTxBytes(t, notaryRequest1.FallbackTransaction)
+			checkGetTxBytes(t, notaryRequest2.FallbackTransaction)
+		})
 	})
 }
 
 // createValidNotaryRequest creates and signs P2PNotaryRequest payload which can
-// pass verification.
-func createValidNotaryRequest(chain *core.Blockchain, sender *keys.PrivateKey, nonce uint32) *payload.P2PNotaryRequest {
+// pass verification. It uses the provided mainTx if it's a nonempty structure.
+func createValidNotaryRequest(chain *core.Blockchain, sender *keys.PrivateKey, nonce uint32, networkFee int64, mainTx *transaction.Transaction) *payload.P2PNotaryRequest {
 	h := chain.BlockHeight()
-	mainTx := &transaction.Transaction{
-		Nonce:           nonce,
-		Attributes:      []transaction.Attribute{{Type: transaction.NotaryAssistedT, Value: &transaction.NotaryAssisted{NKeys: 1}}},
-		Script:          []byte{byte(opcode.RET)},
-		ValidUntilBlock: h + 100,
-		Signers:         []transaction.Signer{{Account: sender.GetScriptHash()}},
-		Scripts: []transaction.Witness{{
-			InvocationScript:   []byte{1, 4, 7},
-			VerificationScript: []byte{3, 6, 9},
-		}},
+	// If mainTx is empty, (at least Nonce == 0) then generate it.
+	if mainTx.Nonce == 0 {
```

Comment:
NACK. Check `if mainTx == nil` and pass nil in those cases where main generation is needed.

Reply:
Fixed.